cmd/nvidia-container-runtime-hook/container_config.go

```diff
@@ -251,7 +224,7 @@ func (hookConfig *hookConfig) getDriverCapabilities(cudaImage image.CUDA, legacy
 func (hookConfig *hookConfig) getNvidiaConfig(image image.CUDA, privileged bool) *nvidiaConfig {
 	legacyImage := image.IsLegacy()
 
-	devices := hookConfig.getDevices(image, privileged)
+	devices := image.GetDevices(hookConfig.AcceptDeviceListAsVolumeMounts, hookConfig.AcceptEnvvarUnprivileged)
```

Comment:
How about also passing these two options to the image constructor. So that we only call: `image.VisibleDevices` here instead of having to pass through the config.

I already had a rough branch from some time ago which I have pushed here: https://github.com/elezar/nvidia-container-toolkit/tree/respect-volume-mounts-gated